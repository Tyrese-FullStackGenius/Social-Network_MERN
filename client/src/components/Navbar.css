.navbar {
  /* background: rgb(47, 21, 102);
  color: rgb(255, 255, 255);
  display: flex;
  justify-content: space-between;
  /* padding: 10px; */
  flex-grow: 1;
}

.menuButton {
  margin-left: -12;
  margin-right: 20;
}

.flex {
  flex-grow: 1;
}
