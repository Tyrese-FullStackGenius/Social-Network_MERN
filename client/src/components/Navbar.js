import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';
import { withStyles } from '@material-ui/core/styles';

import NavbarLeftMenu from './NavbarLeftMenu';
import NavbarRightMenu from './NavbarRightMenu';

const styles = {
  flex: {
    flexGrow: 1
  },
  logo: {
    color: '#fff',
    textDecoration: 'none'
  },
  menuButton: {
    marginLeft: -12,
    marginRight: 20
  },
  root: {
    flexGrow: 1
  }
};

class Navbar extends Component {
  render() {
    const { classes, logoutUser, user } = this.props;
    return (
      <div className={classes.root}>
        <AppBar position="static">
          <Toolbar>
            <IconButton
              className={classes.menuButton}
              color="inherit"
              aria-label="Menu"
            >
              <NavbarLeftMenu user={user} />
            </IconButton>
            <Typography
              className={classes.flex}
              variant="title"
              color="inherit"
            >
              <Link className={classes.logo} to="/">
                MERN Social
              </Link>
            </Typography>
            <div>
              <NavbarRightMenu logoutUser={logoutUser} user={user} />
            </div>
          </Toolbar>
        </AppBar>
      </div>
    );
  }
}

Navbar.propTypes = {
  classes: PropTypes.object.isRequired,
  logoutUser: PropTypes.func.isRequired,
  user: PropTypes.object.isRequired
};

export default withStyles(styles)(Navbar);
