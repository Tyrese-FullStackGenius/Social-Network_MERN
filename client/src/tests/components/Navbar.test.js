import React from 'react';
import { shallow } from 'enzyme';
import Navbar from '../../components/Navbar';

it('should render Navbar', () => {
  const wrapper = shallow(<Navbar logoutUser={jest.fn()} user={{}} />);
  expect(wrapper).toMatchSnapshot();
});
