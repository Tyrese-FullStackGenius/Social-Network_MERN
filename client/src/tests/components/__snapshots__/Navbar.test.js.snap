// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Navbar 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(Navbar)
    logoutUser={[Function]}
    user={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "classes": Object {
        "flex": "Navbar-flex-1",
        "logo": "Navbar-logo-2",
        "menuButton": "Navbar-menuButton-3",
        "root": "Navbar-root-4",
      },
      "logoutUser": [Function],
      "user": Object {},
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "classes": Object {
          "flex": "Navbar-flex-1",
          "logo": "Navbar-logo-2",
          "menuButton": "Navbar-menuButton-3",
          "root": "Navbar-root-4",
        },
        "logoutUser": [Function],
        "user": Object {},
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
